definition/definition.go

```diff
@@ -211,3 +274,33 @@ type Descriptor struct {
 	// Description describes the usage of the path.
 	Description string
 }
+
+// Consume adds consumes.
+func (d Descriptor) Consume(cs ...string) Descriptor {
+	d.Consumes = append(d.Consumes, cs...)
+	return d
+}
+
+// Produce adds produces.
+func (d Descriptor) Produce(ps ...string) Descriptor {
+	d.Produces = append(d.Produces, ps...)
+	return d
+}
+
+// Middleware adds middlewares.
+func (d Descriptor) Middleware(ms ...Middleware) Descriptor {
+	d.Middlewares = append(d.Middlewares, ms...)
+	return d
+}
+
+// Definition adds definitions.
+func (d Descriptor) Definition(defs ...Definition) Descriptor {
+	d.Definitions = append(d.Definitions, defs...)
+	return d
+}
+
+// Descriptor adds descriptors as children.
+func (d Descriptor) Descriptor(descs ...Descriptor) Descriptor {
```

Comment:
I would prefer using `Children` for this function name.

Reply:
```
	v1 := definition.DescriptorFor("/api/v1", "It contains all APIs in v1").
		Consume(definition.MIMEJSON).
		Produce(definition.MIMEJSON).
		Descriptor(descriptors...)
```

vs

```
	v1 := definition.DescriptorFor("/api/v1", "It contains all APIs in v1").
		Consume(definition.MIMEJSON).
		Produce(definition.MIMEJSON).
		Children(descriptors...)
```

The latter is more straightforward.

Reply:
Children is better.  But it is used as struct field.
```go

// Descriptor describes a descriptor for API definitions.
type Descriptor struct {
	// Children is used to place sub-descriptors.
	Children []Descriptor
}
```

So We need another word to the field:
`offspring`, `progeny`, `descendant`

Reply:
Use children?